src/httpserver.cpp

```diff
@@ -93,9 +93,10 @@ class WorkQueue
 
 public:
     WorkQueue(size_t _maxDepth) : running(true),
-                                 maxDepth(_maxDepth),
-                                 numThreads(0)
+                                 maxDepth(_maxDepth)
     {
+        std::lock_guard<std::mutex> lock(cs);
```

Comment:
I'd highly prefer we do - such tools are pretty critical to our ability to find issues - but should probably have a comment noting why we have this insanity.